### PR TITLE
Save restore state to LARA

### PR DESCRIPTION
**Pull request by sfentress**
This creates a serialization and deserialization set of utilities that can prepare the model state to be cleanly serialized to JSON, and the deserialized model can correctly hydrate the model state. This uses the `lara-interactive-api` to handle the actual state saving and loading.

### CODE REVIEW

**Review by mklewandowski — APPROVED** [review]
👍 This looks and works great!  Not much else to add.  Nice move stripping the images out as we discussed!

File: src/components/render-app.tsx [hunk 1]
```diff
@@ -48,8 +49,15 @@ const getState = (): any => {
...
 
-export const renderApp = <IModelInputState, IModelOutputState, IModelConfig>(options: IRenderAppOptions<IModelInputState, IModelOutputState, IModelConfig>) => {
-  const onStateChange: ModelStateChangeCallback<IModelInputState, IModelOutputState> = (state) => currentState = state;
+export const renderApp = <IModelInputState, IModelOutputState, IModelConfig>(
+    options: IRenderAppOptions<IModelInputState, IModelOutputState, IModelConfig>) => {
+  const onStateChange: ModelStateChangeCallback<IModelInputState, IModelOutputState> = (state) => {
+    currentState = state;
+    if (options.onStateChange) {
+      options.onStateChange(state);
+    }
```
Comment (kswenson) [inline src/components/render-app.tsx:58]: These three lines can be simplified to:
```typescript
    options.onStateChange?.(state);
```

File: src/leaf-model-types.ts [hunk 2]
```diff
@@ -27,3 +27,8 @@ export interface IModelOutputState {
 export interface IModelTransientState {
   time: number;
 }
+export interface ISerializableTrayObject extends Omit<TrayObject, "image" | "dragImage"> {}
+export interface ISerializableModelOutputState extends Omit<IModelOutputState, "trayObjects" | "habitatFeatures">  {
+  trayObjects: ISerializableTrayObject[];
+  habitatFeatures: string[];
+}
```
Comment (kswenson) [inline src/leaf-model-types.ts:34]: Nice TypeScript usage!

**Review by kswenson — APPROVED** [review]
👍 Looks good -- nice work!

**Pull request merged**